modules/installing-gcp-user-defined-labels-and-tags.adoc

```diff
@@ -0,0 +1,123 @@
+** An `OrganizationID` must consist of decimal numbers, and cannot have leading zeros.
+** A `ProjectID` must be 6 to 30 characters in length, can only contain lowercase letters, numbers, and hyphens, and must start with a letter, and cannot end with a hyphen.
+* A tag key and value must have a minimum of 1 character and can have a maximum of 63 characters.
+* A tag key must contain only uppercase and lowercase alphanumeric characters, hyphens, underscores, and periods (i.e. ._-).
+* A tag value must contain only uppercase and lowercase alphanumeric characters, hyphens, underscores, periods, at signs, percent signs, equals signs, plusses, colons, commas, asterisks, pound signs, ampersands, parentheses, square braces, curly braces, and spaces (i.e. _-.@%=+:,*#&(){}[] and spaces).
+* A tag key and value must begin and end with an alphanumeric character.
+* Tag value is of type scalar and must be one of the pre-defined values for the key.
+* You can configure a maximum of 50 tags.
+* There should be no tag key with the same value as any of the existing tag keys on a resource(inherited from parent resource).
+
+For more information about identifying the `OrganizationID` or `ProjectID`, see:
+* link:https://cloud.google.com/resource-manager/docs/creating-managing-organization#retrieving_your_organization_id[OrganizationID]
+* link:https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects[ProjectID]
+
+After creating {product-title} cluster, you can access the list of the labels and tags defined for the GCP resources by referring to the new fields `.status.platformStatus.gcp.resourceLabels` and `.status.platformStatus.gcp.resourceTags` present in `infrastructures.config.openshift.io/cluster` object.
+
+.Sample `infrastructure` file
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Infrastructure
+metadata:
+ name: cluster
+spec:
+ platformSpec:
+   type: GCP
+status:
+ infrastructureName: <cluster_id>
+ platform: GCP
+ platformStatus:
+   gcp:
+     resourceLabels:
+     - key: <label_key>
+       value: <label_value>
+     resourceTags:
+     - key: <tag_key_short_name>
+       parentID: <OrganizationID/ProjectID>
+       value: <tag_value_short_name>
+   type: GCP
+----
+
+Along with the user-defined labels, resources will also have a label defined by the {product-title}. The format of the {product-title} labels is `kubernetes-io-cluster-<cluster_id>:owned`. The `cluster_id` parameter is the value of `.status.infrastructureName` present in `infrastructures.config.openshift.io/cluster`.
+
+User-defined and {product-title} labels can be found on the following GCP resoucres:
+
+* Compute Instance
+* Compute Disk
+* Compute Forwarding Rule
+* Compute Image
+* DNS Managed Zone
+* Filestore Instance
+* Storage Bucket
+
+User-defined tags can be found added to the GCP resources, that is, storage bucket.
+
+The following are the limitations for the user-defined tags:
```

Comment:
IMO, this should be a new sub-module.

Reply:
@dfitzmau Do we need to create submodules for both labels and tags?

Reply:
I think one reference module would be enough. This module could capture the folowing lists:

> The user-defined labels have the following requirements:
> The user-defined tags have the following requirements:
> The following are the limitations for the user-defined tags:

I'll leave this choice up to you. I personally found the multiple lists in the concept module difficult to follow, but maybe the limitations list could be stripped out and placed in a reference module or even a separate concept module. Totally up to you.

Reply:
No longer required as sub-heading work really well.

Reply:
+1 @dfitzmau